plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/reload/ReloadPipelinesAction.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.plugin.ingest.transport.reload;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.component.AbstractComponent;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Provider;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.plugin.ingest.PipelineStore;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * Internal api that refreshes the in-memory representation of all the pipelines on all ingest nodes.
+ */
+public class ReloadPipelinesAction extends AbstractComponent implements TransportRequestHandler<ReloadPipelinesAction.ReloadPipelinesRequest> {
+
+    public static final String ACTION_NAME = "internal:admin/ingest/reload/pipelines";
+
+    private final ClusterService clusterService;
+    private final TransportService transportService;
+    private final Provider<PipelineStore> pipelineStoreProvider;
+
+    private PipelineStore pipelineStore;
+
+    @Inject
+    public ReloadPipelinesAction(Settings settings, Provider<PipelineStore> pipelineStoreProvider, ClusterService clusterService, TransportService transportService) {
```

Comment:
so the reason pipeline store instance is managed by Guice is that the transport action classes uses it as a dependency. But that shouldn't be a problem, because if the store is exposed as a getter and the transport action classes depend on `PipelineStoreBootstrapper` then the constructor of the transport action classes can use this getter.